Powershell scripts to support installation from source.

paninaro opened the pull request.

paninaro committed:
Powershell scripts to support installation from source.

paninaro committed:
Update Install-gui.ps1

paninaro committed:
Update Install-gui.ps1

hoffmang9 reviewed (dismissed):
lgtm

harold-b commented on Install-gui.ps1:
```diff
@@ -0,0 +1,33 @@
+git submodule update --init --recursive
+Write-Host "Running git submodule update."
+Write-Host ""
+git submodule update
```
Shouldn't the first `update` call be sufficient?

paninaro replied:
I'm not sure why install-gui.sh has this line. Shouldn't be necessary. I'll remove it from this script.

paninaro committed:
Linter fixes

jack60612 commented:
@paninaro did you mean to include the gui updates?

paninaro commented:
> @paninaro did you mean to include the gui updates?

Ack, thanks for the catch. That slipped past me.

paninaro committed:
Revert "Linter fixes"

This reverts commit 8c4e073671b703ebdab79f30b2f9b68905990698.

paninaro committed:
Linter fixes

jack60612 commented:
@paninaro once you fix the update thing i'll be happy to approve

jack60612 commented:
> @paninaro once you fix the update thing i'll be happy to approve

never mind lol

jack60612 reviewed (dismissed):
LGTM

jack60612 commented:
trying to get this squeezed in for 1.2.8

altendky reviewed (commented):
I don't see any issues, though I have limited PowerShell experience.  All comments here and inline are just things I think might be nice for us or everyone else.

When I am adding suggestions it is weirdly adding an extra blank line into them.  Are these files using `\r\n` newlines?  Do we want that?  I don't know if PowerShell cares, but if not then I personally would stick with `\n`.  Also, apparently it got a BOM added and is thus UTF-8 instead of ASCII?  Doesn't super matter, just a bit odd.  `:]`

```
Install.ps1: UTF-8 Unicode (with BOM) text, with CRLF line terminators
```

A bit of a nitpick, but maybe keeping lowercase names would be good so they sort with the related Linux files?

![image](https://user-images.githubusercontent.com/543719/135547176-2a2d5392-c9c8-41da-bd54-9e4c08a80e7f.png)

paninaro committed:
Minor textual changes. Updated encoding/line-endings to UTF-8/LF. Git on Windows should autocrlf as necessary.

paninaro commented on Install-gui.ps1:
```diff
@@ -0,0 +1,31 @@
+
+if ($null -eq (Get-ChildItem env:VIRTUAL_ENV -ErrorAction SilentlyContinue))
+{
+    Write-Output "This requires the chia python virtual environment."
```
Updated with text similar to the suggestion

paninaro commented on Install-gui.ps1:
```diff
@@ -0,0 +1,31 @@
+Write-Output ""
+Write-Output "Chia blockchain Install-gui.ps1 completed."
+Write-Output ""
+Write-Output "Type 'cd chia-blockchain-gui' and then 'npm run electron' to start the GUI."
```
Done

paninaro commented on Install-gui.ps1:
```diff
@@ -0,0 +1,31 @@
+    Exit 1
+}
+
+Write-Output "Running git submodule update --init --recursive."
```
Done

paninaro commented on Install.ps1:
```diff
@@ -0,0 +1,59 @@
+
+if ($null -eq (Get-Command py -ErrorAction SilentlyContinue))
+{
+    Write-Output "Unable to find py"
```
Done

paninaro commented:
> I don't see any issues, though I have limited PowerShell experience. All comments here and inline are just things I think might be nice for us or everyone else.
> 
> When I am adding suggestions it is weirdly adding an extra blank line into them. Are these files using `\r\n` newlines? Do we want that? I don't know if PowerShell cares, but if not then I personally would stick with `\n`. Also, apparently it got a BOM added and is thus UTF-8 instead of ASCII? Doesn't super matter, just a bit odd. `:]`
> 
> ```
> Install.ps1: UTF-8 Unicode (with BOM) text, with CRLF line terminators
> ```
> 
> A bit of a nitpick, but maybe keeping lowercase names would be good so they sort with the related Linux files?
> 
> ![image](https://user-images.githubusercontent.com/543719/135547176-2a2d5392-c9c8-41da-bd54-9e4c08a80e7f.png)

Updated the encoding to UTF-8 without a BOM. Also changed the line-endings to LF. Git should handle the line ending conversion automatically on checkout.

I kept the naming capitalized as that appears to be a PowerShell convention/best practice.

altendky commented on Install.ps1:
```diff
@@ -0,0 +1,63 @@
+$pythonVersion = (py --version).split(" ")[1]
+if ([version]$pythonVersion -lt [version]"3.7.0")
+{
+    Write-Output "Found Python version:" $pythonVersion
```
I was thinking immediately above the `if` so it was shown in both cases.  Not a big deal though.

jack60612 reviewed (approved):
lgtm
